pkgs/development/cuda-modules/generic-builders/manifest.nix

```diff
@@ -107,15 +109,27 @@ backendStdenv.mkDerivation (
     # Useful for introspecting why something went wrong.
     # Maps descriptions of why the derivation would be marked broken to
     # booleans indicating whether that description is true.
-    brokenConditions = {};
-
-    src = fetchurl {
-      url = "https://developer.download.nvidia.com/compute/${redistName}/redist/${
-        redistribRelease.${redistArch}.relative_path
-      }";
-      inherit (redistribRelease.${redistArch}) sha256;
+    brokenConditions = {
+      # Using an unrecognized redistArch
+      "Unrecognized NixOS platform ${hostPlatform.system}" = redistArch == null;
+      # Trying to build for a platform that doesn't have a redistributable
+      "Unsupported NixOS platform (or configuration) ${hostPlatform.system}" = finalAttrs.src == null;
     };
```

Comment:
The `brokenConditions` you implemented for `torch` had slightly different semantics, which was also closer to the [warnings RFC](https://github.com/NixOS/rfcs/blob/c65c8321782b40844167beb48818471f70900d9d/rfcs/0127-issues-warnings.md#detailed-design): the valid-ness test was just `brokenConditions == {}` 🤔

Reply:
Since other packages need to be able to override this attribute, I do the actual filtering of values in the attribute set inside of `meta`. Otherwise you're looking at overriding the argument given to the filter function rather than the actual attribute set.